cmd/release/github.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v25/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+type app struct {
+	owner  string
+	repo   string
+	client *github.Client
+}
+
+var (
+	contextTimeout = time.Second * 10
+)
+
+func getClient(accessToken string) *github.Client {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{
+			AccessToken: accessToken,
+		},
+	)
+
+	client := github.NewClient(oauth2.NewClient(ctx, ts))
+
+	return client
+}
+
+func new(accessToken string) *app {
+	return &app{
+		owner:  "MeilleursAgents",
+		repo:   "terraform-provider-ansiblevault",
+		client: getClient(accessToken),
+	}
+}
+
+func (a *app) PublishArtifactsOnTag(tag string, buildAssets []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
```

Comment:
Move this declaration into `retriveTag` method. The `defer cancel()` will proceed as soon as the effective call is done, not after all upload of this function

Reply:
i need ctx in `listAssets` and `uploadAssets`

Reply:
Declare a context for each method. Timeout is short

Reply:
ok

Reply:
Done 598e318